plugins/woocommerce-blocks/assets/js/base/components/cart-checkout/form/validate-state.ts

```diff
@@ -0,0 +1,63 @@
+/**
+ * External dependencies
+ */
+import { dispatch, select } from '@wordpress/data';
+import { ShippingAddress } from '@woocommerce/settings';
+import { VALIDATION_STORE_KEY } from '@woocommerce/block-data';
+import { __ } from '@wordpress/i18n';
+
+function previousString( initialValue?: string ) {
+	let lastValue = initialValue;
+
+	function track( value: string ) {
+		const currentValue = lastValue;
+
+		lastValue = value;
+
+		// Return the previous value
+		return currentValue;
+	}
+
+	return track;
+}
+
+const lastValue = previousString( '' );
+
+export const validateState = (
+	addressType: string,
+	values: ShippingAddress,
+	isRequired: boolean
+) => {
+	const validationErrorId = `${ addressType }_state`;
+	const hasValidationError =
+		select( VALIDATION_STORE_KEY ).getValidationError( validationErrorId );
+
+	const countryChanged = lastValue( values.country ) !== values.country;
+
+	if ( hasValidationError ) {
+		if ( ! isRequired || values.state ) {
+			// Validation error has been set, but its no longer required, or the state was provided, clear the error.
```

Comment:
Minor grammar nitpick, sorry 😖 
```suggestion
			// Validation error has been set, but it's no longer required, or the state was provided, clear the error.
```

Reply:
Thanks!